src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,595 @@
+
+        setScopeOfPatternVariableExcludingLineAndColumn(variableDesc, ancestorOfInstanceOfAst);
+    }
+
+    /**
+     * Should scope of pattern variable be updated.
+     *
+     * @param patternVariable DetailAst of type {@link TokenTypes#LITERAL_INSTANCEOF}
+     * @param ancestorOfInstanceOfAst ancestor of {@link TokenTypes#LITERAL_INSTANCEOF} node
+     * @param scopeLineStart starting line of scope
+     * @param scopeLineEnd ending line of scope
+     * @return true if scope of pattern variable should be updated else false
+     */
+    private static boolean shouldScopeBeUpdated(DetailAST patternVariable,
+                                                DetailAST ancestorOfInstanceOfAst,
+                                                int scopeLineStart, int scopeLineEnd) {
+        return !(ancestorOfInstanceOfAst.getType() == TokenTypes.QUESTION
+                && (ancestorOfInstanceOfAst.getLineNo() < scopeLineStart
+                || ancestorOfInstanceOfAst.getLineNo() == scopeLineStart
+                && ancestorOfInstanceOfAst.getColumnNo() < patternVariable.getColumnNo()))
+                && ancestorOfInstanceOfAst.getLineNo() >= scopeLineEnd;
+    }
+
+    /**
+     * Get the updated scopeLineEnd and scopeColumnEnd values.
+     *
+     * @param ancestorOfInstanceOfAst ancestor of {@link TokenTypes#LITERAL_INSTANCEOF} node
+     * @return updated scopeLineEnd and scopeColumnEnd values
+     */
+    private static int[] getUpdatedScopeValues(DetailAST ancestorOfInstanceOfAst) {
+        final int scopeLineEnd;
+        final int scopeColumnEnd;
+        DetailAST childAst = ancestorOfInstanceOfAst.getLastChild();
+
+        // Temporary comment, I feel while(true) condition is more clear,
+        // but if a flag would be considered better, I will modify the condition.
+        while (true) {
+            final DetailAST lastChild = childAst.getLastChild();
+            if (lastChild == null) {
+                break;
+            }
+            childAst = lastChild;
+        }
+        scopeLineEnd = childAst.getLineNo();
+        scopeColumnEnd = childAst.getColumnNo();
+
+        // Additional -1 added to comply with indexes in PatternVariableScope.
+        return new int[] {-1, scopeLineEnd, scopeColumnEnd};
+    }
+
+    /**
+     * Set the scope of pattern variable excluding the information
+     * about its line and column number.
+     *
+     * @param variableDesc information about the local variable
+     * @param ancestorOfInstanceOfAst ancestor of DetailAst of type
+     *                                {@link TokenTypes#LITERAL_INSTANCEOF}
+     */
+    private static void setScopeOfPatternVariableExcludingLineAndColumn(
+            VariableDesc variableDesc, DetailAST ancestorOfInstanceOfAst) {
+
+        DetailAST ast = ancestorOfInstanceOfAst;
+        DetailAST grandParent = ast.getParent();
+        final boolean IsAFullScopeVariable = ast.getType() == TokenTypes.EXPR
```

Comment:
name starts with capital letter, thus ci is failing. Also pls avoid articles in names, just `isFullScopeVariable` or `isTernaryOpExpression` is enough

Reply:
Field not required now